src/run/run.ts

```diff
@@ -24,12 +24,33 @@ export async function run(): Promise<void> {
     let testRunConfigs: TestRunConfig[];
 
     if (trunkMode) {
-      // TODO: Use glob pattern to find the testrail report file
-      // https://github.com/isaacs/node-glob#readme
-      testRunConfigs = await getTrunkTestRunConfigs();
-
-      for (const testRun of testRunConfigs) {
-        await reportToTestrail(closeMilestone, jiraKey, trunkMode, regressionMode, testRun, testRailOptions);
+      const containsE2E = await containsE2Etest();
+      if (!containsE2E && environment === Environment.Production) {
+        testRunConfigs = [await getUnitTestConfig()];
+        const runOptions: RunInputs = {
+          jiraKey,
+          trunkMode,
+          regressionMode,
+          testRunConfig: testRunConfigs[0],
+        };
+        const testrailService = new TestrailService(testRailOptions, runOptions);
+        // get the milestone created in the build/test-unit job
+        const milestone = await testrailService.establishMilestone().catch((error) => {
+          setFailed("A TestRail Milestone could not be established.");
+          throw error;
+        });
+        // and close it
+        await testrailService.closeMilestone(milestone.id).catch((error) => {
+          setFailed("The TestRail Milestone could not be closed.");
+          throw error;
+        });
```

Comment:
I think it would be a lot cleaner to separate this into its own function and give it a good name.

Reply:
Done in https://github.com/q4mobile/stratus-testrail-reporter/pull/33/commits/42ff252d20ab132dc2b40690c3c5fc8094623467.